server/src/main/java/org/apache/druid/server/coordinator/loading/StrategicSegmentAssigner.java

```diff
@@ -446,7 +446,8 @@ private int dropReplicas(
     for (ServerHolder server : eligibleServers) {
       if (server.isDecommissioning()) {
         eligibleDyingServers.add(server);
-      } else {
+      } else if (!server.isUnmanaged()) {
```

Comment:
Rather than having this check in all places, we should keep a separate set of `managedHistoricals` in `DruidCluster`, and add a method `getManagedHistoricals()` and all places that do segment assignment, balancing etc, should use `getManagedHistoricals()` instead of `getHistoricals()`.

Similarly, you would also need to keep a set of `allManagedServers`.
Please let me know if you need any further clarification on this.

Reply:
1. Is `allManagedServers` required? I don't see any use current use case of `getAllServers()` which should be replaced.
2. Should we add a different metric CLONE_COUNT instead of HISTORICAL_COUNT for clone servers to track capacity better? Should clones affect `tier/total/capacity`?

Reply:
1. There are two usages of `allServers()` that need to be removed. One is `StrategicSegmentAssigner.broadcast()`. The other is in `UnloadUnusedSegments`.
2. We can leave historical count metric as is. But you can emit a clone count in addition to that to track the number of clones. The clones should not count towards the tier capacity.

Reply:
Change. Please take a look.